tests/containers_test.go

```diff
@@ -160,6 +160,137 @@ func TestCLIToolTests(t *testing.T) {
 	})
 }
 
+func TestEnvironment(t *testing.T) {
+	t.Parallel()
+	// Deployment steps are run via bash, and explicitly set BASH_ENV to source ~/.bashrc.
+	// https://github.com/pulumi/pulumi-service/blob/8cbd9397ec0cdc7b5c168715ca4c9aa087c83823/cmd/workflow-runner/run.go#L78
+	// We run commands that check basic assertions about the environment within the container once
+	// directly without shell and once with bash to ensure that the environment is set up correctly
+	// for both cases.
+	// This is a regression test for https://github.com/pulumi/pulumi-docker-containers/issues/193
+
+	imageVariant := os.Getenv("IMAGE_VARIANT")
+	t.Logf("Testing image variant: %s", imageVariant)
+
+	t.Run("Python", func(t *testing.T) {
+		if !strings.HasSuffix(imageVariant, "python") {
+			t.Skip("Skipping test for non python images")
+		}
+		t.Parallel()
+		expected := "/usr/local/bin/python"
+		if imageVariant == "pulumi-ubi-python" {
+			expected = "/usr/bin/python"
+		}
+
+		p, err := exec.LookPath("python")
+		require.NoError(t, err)
+		require.Equal(t, expected, p)
+		// Use bash `command` buultin to lookup the path to python
+		requireOutputWithBash(t, expected, "command", "-v", "python")
+	})
+
+	t.Run("Node", func(t *testing.T) {
+		if !strings.HasSuffix(imageVariant, "node") {
+			t.Skip("Skipping test for non node images")
+		}
+		t.Parallel()
+		p, err := exec.LookPath("python")
```

Comment:
Are we supposed to look up node here rather than python?  Why is this test succeeding on the node image?

Reply:
Oops ... this was always skipped because the suffix for node images is `nodejs`, not `node`